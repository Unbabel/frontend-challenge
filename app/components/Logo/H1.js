import styled from 'styled-components';

export default styled.h1`
  color: ${props => props.theme.titleColor};
  font-size: 18px;
  font-weight: 500;
  line-height: 22px;
  overflow: hidden;
  outline: none;
  text-decoration: none;
`;
