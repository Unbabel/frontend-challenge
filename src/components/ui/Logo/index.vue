<template>
  <a href="/">
    <h1>
      <span>T</span>
      <span>r</span>
      <span>a</span>
      <span>n</span>
      <span>s</span>
      <span>c</span>
      <span>r</span>
      <span>i</span>
      <span>p</span>
      <span>t</span>
      <span>i</span>
      <span>o</span>
      <span>n</span>
      <span>s</span>
    </h1>
  </a>
</template>
<script>
export default {
  name: "Logo",
};
</script>
<style scoped lang="scss">
h1 {
  color: $color-header-title;
  font-family: $font-montserrat;
  font-size: 18px;
  font-weight: 500;
  line-height: 22px;
  overflow: hidden;
  outline: none;
  text-decoration: none;
  position: relative;
  &::before {
    content: '';
    position: absolute;
    height: 100%;
    width: 100%;
    left: 0;
    transform: translate3d(-101%, 0, 0);
    transition: transform 0.5s;
  }
  &:hover::before {
    transform: translate3d(0, 0, 0);
  }
  span {
    display: inline-block;
    position: relative;
    transform: perspective(1000px) rotate3d(0, 1, 0, 0deg);
    transition: transform 0.5s, color 0.5s;
  }
  &:hover span {
    color: $color-purple;
    transform: perspective(1000px) rotate3d(0, 1, 0, 360deg);
  }
  span:nth-child(14),
  &:hover span:first-child {
    transition-delay: 0s;
  }
  span:nth-child(13),
  &:hover span:nth-child(2) {
    transition-delay: 0.05s;
  }
  span:nth-child(12),
  &:hover span:nth-child(3) {
    transition-delay: 0.1s;
  }
  span:nth-child(11),
  &:hover span:nth-child(4) {
    transition-delay: 0.15s;
  }
  span:nth-child(10),
  &:hover span:nth-child(5) {
    transition-delay: 0.2s;
  }
  span:nth-child(9),
  &:hover span:nth-child(6) {
    transition-delay: 0.25s;
  }
  span:nth-child(8),
  &:hover span:nth-child(7) {
    transition-delay: 0.3s;
  }
  span:nth-child(7),
  &:hover span:nth-child(8) {
    transition-delay: 0.35s;
  }
  span:nth-child(6),
  &:hover span:nth-child(9) {
    transition-delay: 0.4s;
  }
  span:nth-child(5),
  &:hover span:nth-child(10) {
    transition-delay: 0.45s;
  }
  span:nth-child(4),
  &:hover span:nth-child(11) {
    transition-delay: 0.5s;
  }
  span:nth-child(3),
  &:hover span:nth-child(12) {
    transition-delay: 0.55s;
  }
  span:nth-child(2),
  &:hover span:nth-child(13) {
    transition-delay: 0.6s;
  }
  span:first-child,
  &:hover span:nth-child(14) {
    transition-delay: 0.65s;
  }
}
</style>
