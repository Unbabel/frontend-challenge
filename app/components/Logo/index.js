import React, { useContext } from 'react';
import { ThemeContext } from 'styled-components';

import H1 from './H1';

function Logo() {
  return <H1 theme={useContext(ThemeContext)}>Transcriptions</H1>;
}

export default Logo;
