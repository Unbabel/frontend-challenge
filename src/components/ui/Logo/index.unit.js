import { shallowMount } from "@vue/test-utils";

import Logo from "./index.vue";

describe("@/components/ui/Logo", () => {
  it("renders anchor tag", () => {
    const wrapper = shallowMount(Logo);
    expect(wrapper.contains("router-link")).toBe(true);
  });
  it("renders h1 tag", () => {
    const wrapper = shallowMount(Logo);
    expect(wrapper.contains("h1")).toBe(true);
  });
});
