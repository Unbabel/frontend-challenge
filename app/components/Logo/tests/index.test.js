import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { IntlProvider } from 'react-intl';
import { ConnectedRouter } from 'connected-react-router/immutable';
import { createMemoryHistory } from 'history';

import Logo from '../index';
import configureStore from '../../../configureStore';

describe('<Logo />', () => {
  const history = createMemoryHistory();
  const store = configureStore({}, history);

  it('should render a h1', () => {
    const { container } = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <ConnectedRouter history={history}>
            <Logo />
          </ConnectedRouter>
        </IntlProvider>
      </Provider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
