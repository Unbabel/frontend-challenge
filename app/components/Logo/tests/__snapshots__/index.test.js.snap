// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> should render a div 1`] = `
<h1
  class="H1-sc-1uc3xdc-0 jfHfyl"
>
  <span>
    T
  </span>
  <span>
    r
  </span>
  <span>
    a
  </span>
  <span>
    n
  </span>
  <span>
    s
  </span>
  <span>
    c
  </span>
  <span>
    r
  </span>
  <span>
    i
  </span>
  <span>
    p
  </span>
  <span>
    t
  </span>
  <span>
    i
  </span>
  <span>
    o
  </span>
  <span>
    n
  </span>
  <span>
    s
  </span>
</h1>
`;

exports[`<Logo /> should render a h1 1`] = `
<h1
  class="H1-sc-1uc3xdc-0 jfHfyl"
>
  <span>
    T
  </span>
  <span>
    r
  </span>
  <span>
    a
  </span>
  <span>
    n
  </span>
  <span>
    s
  </span>
  <span>
    c
  </span>
  <span>
    r
  </span>
  <span>
    i
  </span>
  <span>
    p
  </span>
  <span>
    t
  </span>
  <span>
    i
  </span>
  <span>
    o
  </span>
  <span>
    n
  </span>
  <span>
    s
  </span>
</h1>
`;
